pinot-core/src/main/java/com/linkedin/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -143,6 +147,22 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
         metadata.put(DataTable.NUM_ENTRIES_SCANNED_POST_FILTER_METADATA_KEY, "0");
         metadata.put(DataTable.NUM_SEGMENTS_PROCESSED, "0");
         metadata.put(DataTable.NUM_SEGMENTS_MATCHED, "0");
+
+        // adds pruning reasons to metadata.
+        StringBuilder pruningReasons = new StringBuilder();
+        for (SegmentPruner segmentPruner : matchedSegmentPruners) {
+          // emits metric when querying non-existing columns.
+          if ("DataSchemaSegmentPruner".equals(segmentPruner.toString())) {
```

Comment:
This seems a bit involved to surface the information we are looking for. Can this be handled better on the broker side? Broker can cache the schema and check for columns queried against the schema?

Reply:
Yeah, putting the check to broker side seems to be the right place. 
In that case, `DataSchemaSegmentPruner` would be useless.